modules/solid_properties/src/materials/ThermalSolidPropertiesFunctorMaterial.C

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ThermalSolidPropertiesFunctorMaterial.h"
+#include "ThermalSolidProperties.h"
+#include "SolidPropertiesNames.h"
+
+registerMooseObject("SolidPropertiesApp", ThermalSolidPropertiesFunctorMaterial);
+
+InputParameters
+ThermalSolidPropertiesFunctorMaterial::validParams()
+{
+  InputParameters params = FunctorMaterial::validParams();
+  params.addRequiredParam<MooseFunctorName>("temperature", "Temperature");
+  params.addRequiredParam<UserObjectName>("sp", "The name of the user object for solid properties");
+  params.addParam<std::string>(SolidPropertiesNames::specific_heat,
+                               SolidPropertiesNames::specific_heat,
+                               "Name to be used for the isobaric specific heat");
+  params.addParam<std::string>(SolidPropertiesNames::thermal_conductivity,
+                               SolidPropertiesNames::thermal_conductivity,
+                               "Name to be used for the thermal conductivity");
+  params.addParam<std::string>(SolidPropertiesNames::density,
+                               SolidPropertiesNames::density,
+                               "Name to be used for the density");
+  params.addClassDescription("Computes solid thermal properties as a function of temperature");
+  return params;
+}
+
+ThermalSolidPropertiesFunctorMaterial::ThermalSolidPropertiesFunctorMaterial(
+    const InputParameters & parameters)
+  : FunctorMaterial(parameters),
+    _temperature(getFunctor<ADReal>("temperature")),
+    _sp(getUserObject<ThermalSolidProperties>("sp"))
+{
+  addFunctorProperty<ADReal>(getParam<std::string>(SolidPropertiesNames::specific_heat),
```

Comment:
given the new conversion capability I dont think there s a need to template those on AD anymore